src/modules/workflows/workflowEngine.js

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable indent */
+const { Status, WorkflowKeysEnum } = require('./enums');
+
+module.exports = (workflow, { writer, editor }) => {
+  const currentStatus = workflow.status;
+  const changeLog = {};
+
+  switch (currentStatus) {
+    case Status.WAITING_FOR_WRITER:
```

Comment:
this is the very first status that will be assigned to a workflow (during creation).
While creating, we won't have the writer's information.
So, this error will be thrown always.
is that the expectation ?